charts/newrelic-infrastructure/templates/controlplane/daemonset.yaml

```diff
@@ -129,6 +129,12 @@ spec:
                   apiVersion: "v1"
                   fieldPath: "spec.nodeName"
 
+            - name: "NRIA_OVERRIDE_HOSTNAME"
+              valueFrom:
+                fieldRef:
+                  apiVersion: "v1"
+                  fieldPath: "spec.nodeName"
+
```

Comment:
I would ask for review to @newrelic/caos here. Bur while we decide if we would have to ask them or not this change does not seem correct. because we still have `NRIA_OVERRIDE_HOSTNAME_SHORT`.

We should only need `NRIA_OVERRIDE_HOSTNAME` for this change. We may also want to disable (or do a test) [disabling DNS resolution](https://github.com/newrelic/infrastructure-agent/blob/6832e08d7bf4fdb4589138cf88ac0bd071f6f445/pkg/config/config.go#L195-L200): 
```yaml
// DnsHostnameResolution When true, the full hostname is resolved by performing a reverse lookup of the hosts
// address; otherwise, it will be retrieved with the hostname command on Linux, and from the TCP/IP parameters of
// the registry on Windows.
// Default: True
// Public: Yes
DnsHostnameResolution bool `yaml:"dns_hostname_resolution" envconfig:"dns_hostname_resolution"`
```

Reply:
- The `NRIA_OVERRIDE_HOSTNAME_SHORT` is still needed even if you specify `NRIA_OVERRIDE_HOSTNAME` otherwise the short hostname get resolved.
<img width="1975" alt="Screen Shot 2022-06-14 at 4 00 53 PM" src="https://user-images.githubusercontent.com/43335750/173595969-6dac30a4-7e3a-45e7-aea2-fc2553ee915d.png">

 - However, it is possible that `DnsHostnameResolution=false` is enough. But as far as I understand it gets the result of `hostname` command for both full and short, that is the name of the pod. But I might have misunderstood you
<img width="2020" alt="Screen Shot 2022-06-14 at 4 00 03 PM" src="https://user-images.githubusercontent.com/43335750/173595785-c186cbe1-f4a5-4230-8f59-5cebc2550683.png">

Reply:
Overriding both *and* disabling the DNS resolution could be useful to reduce some calls and save some cpu. But as far as I understood from the code if hostname are overwritten no resolution is performed. But @rubenruizdegauna can clarify if it is better to disable them anyway
<img width="1923" alt="Screen Shot 2022-06-14 at 4 05 38 PM" src="https://user-images.githubusercontent.com/43335750/173597024-5f12cc90-680f-48a9-a6c5-89d3d88cd3a5.png">

Reply:
If both short and full are overridden, then no DNS resolution will be made, so I would stick to these.